pandas/_libs/tslibs/offsets.pyx

```diff
@@ -206,3 +271,109 @@ class ApplyTypeError(TypeError):
 # TODO: unused.  remove?
 class CacheableOffset(object):
     _cacheable = True
+
+
+class BeginMixin(object):
```

Comment:
BeginMixin and EndMixin are new, each only have the one method.  At the moment these methods are in DateOffset, but they are only used by a small handful of FooBegin and BarEnd subclasses.